test/e2e/network_test.go

```diff
@@ -574,6 +574,28 @@ func (tc *testContext) getPodEvents(name string) ([]v1.Event, error) {
 	return podEvents, nil
 }
 
+// createLinuxCurlerJob creates a linux job to curl a specific endpoint. curl must be present in the container image.
+func (testCtx *testContext) createLinuxCurlerJob(jobSuffix, endpoint string, continuous bool) (*batchv1.Job, error) {
+	// Retries a failed curl attempt once to avoid flakes
+	curlCommand := fmt.Sprintf(
+		"curl %s;"+
+			" if [ $? != 0 ]; then"+
+			" sleep 60;"+
+			" curl %s || exit 1;"+
```

Comment:
You are still calling `curl` twice here. Can't you use `break` or `continue`?

Reply:
In other words, always have the [while loop](https://linuxize.com/post/bash-while-loop/).

Reply:
the original intent of the 2 curls was to error on the second failure. I'll see if a do-while loop will allow me to remove the second one

Reply:
You can use [this](https://www.networkworld.com/article/3541298/how-to-repeat-a-linux-command-until-it-succeeds.html) for inspiration.

Reply:
After exploring this further I lean towards the current approach. To make it only 1 curl call we'd have to introduce a counter and it looks more complex than the current code.

Reply:
Yeah I was thinking the counter would be more clear but I will defer to your judgement on this. Do we really need the continuous option? Won't doing the loop work in all cases?